include/cantera/zeroD/FlowDeviceFactory.h

```diff
@@ -0,0 +1,61 @@
+//! @file FlowDeviceFactory.h
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef FLOWDEVICE_FACTORY_H
+#define FLOWDEVICE_FACTORY_H
+
+#include "cantera/base/FactoryBase.h"
+#include "cantera/zeroD/FlowDevice.h"
+
+namespace Cantera
+{
+
+class FlowDeviceFactory : public Factory<FlowDevice>
+{
+public:
+    static FlowDeviceFactory* factory() {
+        std::unique_lock<std::mutex> lock(flowDevice_mutex);
+        if (!s_factory) {
+            s_factory = new FlowDeviceFactory;
+        }
+        return s_factory;
+    }
+
+    virtual void deleteFactory() {
+        std::unique_lock<std::mutex> lock(flowDevice_mutex);
+        delete s_factory;
+        s_factory = 0;
+    }
+
+    /**
+     * Create a new FlowDevice.
+     * @param n the type to be created.
+     */
+    virtual FlowDevice* newFlowDevice(int n);
+    virtual FlowDevice* newFlowDevice(const std::string& flowDeviceType);
+
+    //! Register a new object type
+    void reg_type(const std::string& name, const int type) {
```

Comment:
This function could use a bit more documentation, to explain that to register a new type, you must call the `reg` method and assign a string identifier, and that calling this method is then necessary only if you need to assign an integer identifier for creating this type from clib or Matlab.

Reply:
done.